rholang/src/main/scala/coop/rchain/rholang/interpreter/normalize.scala

```diff
@@ -617,60 +634,54 @@ object ProcNormalizeMatcher {
         normalizeMatch[M](p.proc_, input)
 
       case p: PSend =>
-        containsConnective(p.listproc_) match {
-          case Some(errMsg) =>
-            sync.raiseError(
-              SendDataConnectivesNotAllowedError(errMsg)
-            )
-          case None =>
-            for {
-              nameMatchResult <- NameNormalizeMatcher.normalizeMatch[M](
-                                  p.name_,
-                                  NameVisitInputs(input.env, input.knownFree)
-                                )
-              initAcc = (
-                Vector[Par](),
-                ProcVisitInputs(VectorPar(), input.env, nameMatchResult.knownFree),
-                BitSet(),
-                false
-              )
-
-              dataResults <- p.listproc_.toList.reverse.foldM(initAcc)(
-                              (acc, e) => {
-                                normalizeMatch[M](e, acc._2).map(
-                                  procMatchResult =>
-                                    (
-                                      procMatchResult.par +: acc._1,
-                                      ProcVisitInputs(
-                                        VectorPar(),
-                                        input.env,
-                                        procMatchResult.knownFree
-                                      ),
-                                      acc._3 | procMatchResult.par.locallyFree,
-                                      acc._4 || procMatchResult.par.connectiveUsed
-                                    )
+        for {
+          _ <- failOnConnectivesUsed(p.listproc_, SendDataConnectivesNotAllowedError(_))
```

Comment:
This shouldn't be unconditional. You should only fail if you are at the top-level, which is input.env.depth == 0